tests/conftest.py

```diff
@@ -137,3 +180,332 @@ async def client(monkeypatch, mock_openai_chatcompletion, mock_openai_embedding,
+            account_url=None, file_system_name=None, credential=None
+        )
+        return self.filesystems[name]
+
+    monkeypatch.setattr(azure.storage.filedatalake.DataLakeServiceClient, "__init__", mock_init)
+    monkeypatch.setattr(
+        azure.storage.filedatalake.DataLakeServiceClient, "get_file_system_client", mock_get_file_system_client
+    )
+
+    monkeypatch.setattr(azure.storage.filedatalake.aio.DataLakeServiceClient, "__init__", mock_init_service_client_aio)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.DataLakeServiceClient, "__aenter__", mock_aenter)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.DataLakeServiceClient, "__aexit__", mock_aexit)
+    monkeypatch.setattr(
+        azure.storage.filedatalake.aio.DataLakeServiceClient, "get_file_system_client", mock_get_file_system_client_aio
+    )
+
+    def mock_init_filesystem_aio(self, *args, **kwargs):
+        self.directories = {}
+
+    def mock_get_file_client(self, path, *args, **kwargs):
+        return azure.storage.filedatalake.DataLakeFileClient(
+            account_url=None, file_system_name=None, file_path=path, credential=None
+        )
+
+    async def mock_exists_aio(self, *args, **kwargs):
+        return False
+
+    async def mock_create_filesystem_aio(self, *args, **kwargs):
+        pass
+
+    async def mock_create_directory_aio(self, directory, *args, **kwargs):
+        if directory in self.directories:
+            return self.directories[directory]
+        self.directories[directory] = azure.storage.filedatalake.aio.DataLakeDirectoryClient(directory)
+        return self.directories[directory]
+
+    def mock_get_root_directory_client_aio(self, *args, **kwargs):
+        if "/" in self.directories:
+            return self.directories["/"]
+        self.directories["/"] = azure.storage.filedatalake.aio.DataLakeDirectoryClient("/")
+        self.directories["/"].child_directories = self.directories
+        return self.directories["/"]
+
+    def mock_get_paths(self, *args, **kwargs):
+        return [argparse.Namespace(is_directory=False, name=name) for name in ["a.txt", "b.txt", "c.txt"]]
+
+    monkeypatch.setattr(azure.storage.filedatalake.FileSystemClient, "__init__", mock_init)
+    monkeypatch.setattr(azure.storage.filedatalake.FileSystemClient, "get_file_client", mock_get_file_client)
+    monkeypatch.setattr(azure.storage.filedatalake.FileSystemClient, "get_paths", mock_get_paths)
+
+    monkeypatch.setattr(azure.storage.filedatalake.aio.FileSystemClient, "__init__", mock_init_filesystem_aio)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.FileSystemClient, "__aenter__", mock_aenter)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.FileSystemClient, "__aexit__", mock_aexit)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.FileSystemClient, "get_paths", mock_get_paths)
+    monkeypatch.setattr(azure.storage.filedatalake.aio.FileSystemClient, "exists", mock_exists_aio)
```

Comment:
Are these monkeypatch's being done in the global namespace? (As opposed to being in a fixture?) I think we always want to do it in a fixture, to have more predictability about when things will be monkeypatched.

Reply:
ah shoot i need to add some tabs. good catch! should have been in this fixture

@pytest.fixture
def mock_data_lake_service_client

Reply:
it actually does look like there are tabs in there, i think it was just not looking like it how github rendered. But it's under a fixture.